api/desecapi/tests/test_user_management.py

```diff
@@ -839,6 +858,156 @@ def test_reset_password_no_code_override(self):
         self._test_reset_password(self.email, password, code='foobar')
         self.assertPassword(self.email, password)
 
+    def test_renew_domain_non_local_public_child(self):
+        user = User.objects.get(email=self.email)
+        domain = self.create_domain(owner=user)
+        for days in [182, 184]:
+            domain.published = timezone.now() - timedelta(days=days)
+            for renewal_changed in [None, domain.published]:
+                domain.renewal_changed = renewal_changed
+                domain.save()
+
+                self.assertEqual(Domain.objects.get(pk=domain.pk).renewal_state, Domain.RenewalState.FRESH)
+                call_command('scavenge-unused')
+                self.assertEqual(Domain.objects.get(pk=domain.pk).renewal_state, Domain.RenewalState.FRESH)
+                self.assertEqual(Domain.objects.get(pk=domain.pk).renewal_changed, renewal_changed)
+                self.assertEqual(Domain.objects.get(pk=domain.pk).published, domain.published)
+                self.assertEqual(len(mail.outbox), 0)
+
+    def test_renew_domain_recently_published(self):
+        PublicSuffixMockMixin.setUpMockPatch(self)
+        local_public_suffix = random.sample(self.AUTO_DELEGATION_DOMAINS, 1)[0]
```

Comment:
`random.choice`? :)

Reply:
haha... had copied it from elsewhere in our test code, which apparently is older than Python 3.6 (which introduced `random.choice`)

Reply:
ahh... unfortunately, that does not work, because `random.choice` requires a sequence but `self.AUTO_DELEGATION_DOMAINS` is a set. I tried making that class attribute a sequence (by wrapping in `tuple`), but that breaks other code.